datafusion/functions-aggregate/src/array_agg.rs

```diff
@@ -623,8 +668,11 @@ impl Accumulator for OrderSensitiveArrayAggAccumulator {
         let mut partition_ordering_values = vec![];
 
         // Existing values should be merged also.
-        partition_values.push(self.values.clone().into());
-        partition_ordering_values.push(self.ordering_values.clone().into());
+        if !self.is_input_pre_ordered {
```

Comment:
does this have the effect of sorting all inputs even when `ARRAY_AGG` didn't explicitly call for ordering?